src/datamodule.py

```diff
@@ -1,33 +1,33 @@
 """
-LightningDataModule to loads Earth Observation data from <file format> using
-<library>.
+LightningDataModule to load Earth Observation data from GeoTIFF files using
+rasterio.
 """
 import lightning as L
+import numpy as np
+import rasterio
 import torch
+import torchdata
 
 
 # %%
-class RandomDataset(torch.utils.data.Dataset):
+def _array_to_torch(filepath: str) -> torch.Tensor:
     """
-    Torch Dataset that returns tensors of size (13, 256, 256) with random
-    values.
+    Read a GeoTIFF file using rasterio into a numpy.ndarray, and convert it
+    to a torch.Tensor (float16 dtype).
     """
+    # GeoTIFF - Rasterio
+    with rasterio.open(fp=filepath) as dataset:
```

Comment:
Did you experiment with other file openers? Maybe the loader gets more stable if we use 
```
from skimage import io
im = io.imread(filepath)
```

or other tif specific loaders like

https://pypi.org/project/tifffile/

Could be worth a shot to see if that helps stabilizing the loader when compared to zarr.

Reply:
So `skimage`'s imread actually uses `tifffile` behind the scenes for reading TIFF files, see https://github.com/scikit-image/scikit-image/blob/441fe68b95a86d4ae2a351311a0c39a4232b6521/skimage/io/_io.py#L16-L68. I'll need to see if skimage/tifffile supports reading from s3 buckets directly like rasterio/GDAL. Found this thread https://github.com/cgohlke/tifffile/issues/125 which looks interesting.